### PR TITLE
Use `cupy.linalg.det` in `F.det`

### PR DESCRIPTION
**Pull request by toslunar**
This PR removes internal functions that are already available in cupy.

### CODE REVIEW

File: chainer/functions/math/erf.py [hunk 1]
```diff
@@ -39,11 +39,7 @@ def forward_cpu(self, x):
 
     def forward_gpu(self, x):
         self.retain_inputs((0,))
-        return cuda.elementwise(
-            'T x', 'T y',
-            'y = erf(x)',
-            'elementwise_erf',
-        )(x[0]),
+        return cuda.cupyx.scipy.special.erf(x[0]).astype(x[0].dtype),
```
Comment (kmaehashi) [inline chainer/functions/math/erf.py]: `copy=False`

**Comment by kmaehashi** [comment]
Benchmark results are as follows:

```$ asv run --python=same --show-stderr -b ErfFunc.time_forward```
https://github.com/chainer/chainer-benchmark/pull/29

## master branch

```
========= =============== ===============
--                     dtype
--------- -------------------------------
 backend   numpy.float16   numpy.float32
========= =============== ===============
   gpu        905.69μs        973.24μs
   cpu        472.15ms        297.40ms
========= =============== ===============
```

## This PR (rebased to master and fixed copy=False issue):

```
========= =============== ===============
--                     dtype             
--------- -------------------------------
 backend   numpy.float16   numpy.float32 
========= =============== ===============
   gpu         1.58ms         976.33μs   
   cpu        470.99ms        296.67ms   
========= =============== ===============
```

**Comment by toslunar** [comment]
I reverted the changes to `F.erf*` because the slowdown with `float16` doesn't seem fixable.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by kmaehashi** [comment]
LGTM!

**Pull request merged**